pkg/vm/ch.go

```diff
@@ -61,6 +62,12 @@ func (m *Machine) Run(ctx context.Context, socket, logs string) error {
 	if len(filesystems) > 0 {
 		args["--fs"] = filesystems
 	}
+	if m.Corex {
```

Comment:
I think this code should not be part of `vmd` instead in the provision. Because this should only work in `container` mode. Hence provisiond (vm handler) can decide if it's possible or not to run corex. then it can override the entry point (also copy corex binary) before calling vmd which then has to do nothing.